src/stub/parser.rs

```diff
@@ -1,32 +1,29 @@
 #![allow(clippy::while_let_on_iterator)]
 
 pub mod types;
+
 pub use types::{Cmd, JoinTerm, JoinTermType, Stub, VariableCommand};
-mod token_stream;
 
 use regex::Regex;
-use token_stream::TokenStream;
+use std::iter;
 
 pub fn parse_generator_stub(generator: &str) -> Stub {
-    Parser::from(generator).parse()
+    Parser::new(generator).parse()
 }
 
 /// A wrapper around an iterator of tokens in the CG stub. Contains all of the stub parsing logic.
 ///
 /// Exists solely to be consumed with `.parse()`
 pub struct Parser<'a> {
-    token_stream: TokenStream<'a>,
+    token_stream: Box<dyn Iterator<Item = &'a str> + 'a>,
 }
 
-impl<'a> From<&'a str> for Parser<'a> {
-    fn from(value: &'a str) -> Self {
-        let token_stream = TokenStream::from(value);
-        Self { token_stream }
+impl<'a> Parser<'a> {
+    pub fn new(stub: &'a str) -> Self {
+        let token_stream = stub.lines().flat_map(|line| line.split(' ').chain(iter::once("\n")));
```

Comment:
@daxida thoughts?

Reply:
P. S.: Chain just adds an iterator to the end of another one, and `once` creates an iterator out of a single element. Unfortunately I cannot concat `&str`s which would have made this much simpler.